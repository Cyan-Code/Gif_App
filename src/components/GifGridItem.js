import React from 'react'

export const GifGridItem = ( img ) => {

  return (
    <div className="card animate__fadeIn">
      <p> {img.title} </p>
      <img src={img.url} alt={img.title} />
    </div>
  )
}