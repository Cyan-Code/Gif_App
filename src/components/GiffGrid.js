import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GiffGrid = ({ category }) => {

  const {data, loading} = useFetchGifs(category)

  return (
    <>
      <h3 className="card animate__fadeIn"> {category} </h3>
      
      { loading && <p className = "card animate__flash">Loading</p>}

      <div className="card-grid">
        <ol>
          <li>
            {
              data.map ( img => (
                <GifGridItem
                  key = { img.id }
                  { ...img }
                />
              ))
            }
          </li>
        </ol>
      </div>
    </>
  )
}

GiffGrid.propTypes = {
  category: PropTypes.string.isRequired
}
