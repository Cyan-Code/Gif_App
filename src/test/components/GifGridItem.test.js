import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('GifGridItem renderizado correctamente', () => {
  
  const title = 'un titulo'
  const url = 'https://localhost/algo.jpg'
  const wrapper = shallow( <GifGridItem  title = {title} url = {url}/> )
  
  test('Debe renderizar el componentes de manera automÃ¡tica', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de tener un parrafo con el title', () => {
    const p = wrapper.find('p')
    expect( p.text().trim() ).toBe (title)  
  })
  
  test('Debe tener el src y el titulo', () => {
    const img = wrapper.find('img')

    expect( img.prop('src')).toBe(url)
    expect( img.prop('alt')).toBe(title)

  })

  test('Debe de tener animate__fadeIn', () => {
    const div = wrapper.find('div')
    const className = div.prop('className')
    expect( className.includes('animate__fadeIn')).toBe(true)
  })
  
})

