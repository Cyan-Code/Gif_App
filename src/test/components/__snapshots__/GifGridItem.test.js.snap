// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem renderizado correctamente Debe renderizar el componentes de manera autom√°tica 1`] = `
<div
  className="card animate__animate animate__fadeIn"
>
  <p>
     
    un titulo
     
  </p>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
